### PR TITLE
Pods 7394

### PR DESCRIPTION
**Pull request by grantrobson**

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -3,9 +3,10 @@
...
 GET        /versions                                controllers.EventReportController.getVersions
 
-POST       /save                                    controllers.EventReportController.saveEvent
+GET        /event                                   controllers.EventReportController.getEvent
 POST       /compile                                 controllers.EventReportController.compileEvent
 
 POST       /submit-event-declaration-report         controllers.EventReportController.submitEventDeclarationReport
 
-GET        /event                                   controllers.EventReportController.getEvent
+GET        /cache/user-answers                      controllers.EventReportController.getUserAnswersFromCache
```
Comment (cyeniceri) [inline conf/app.routes:11]: I think we shouldn't expose our backend structure as /cache/..

would it be better if we say only /user-answers ? 

Method name can stay as is.

File: conf/app.routes [hunk 2]
```diff
@@ -3,9 +3,10 @@
...
+POST       /cache/user-answers                      controllers.EventReportController.saveUserAnswersToCache
```
Comment (cyeniceri) [inline conf/app.routes:12]: same as above

File: app/controllers/EventReportController.scala [hunk 3]
```diff
@@ -46,13 +46,28 @@ class EventReportController @Inject()(
...
 
-  def saveEvent: Action[AnyContent] = Action.async {
+  def saveUserAnswersToCache: Action[AnyContent] = Action.async {
     implicit request =>
       withPstrEventTypeAndBody { (pstr, eventType, userAnswersJson) =>
         logger.debug(message = s"[Save Event: Incoming-Payload]$userAnswersJson")
         EventType.getEventType(eventType) match {
           case Some(et) =>
-            eventReportService.saveEvent(pstr, et, userAnswersJson).map(_ => Created)
+            eventReportService.saveEventToMongo(pstr, et, userAnswersJson).map(_ => Ok)
```
Comment (cyeniceri) [inline app/controllers/EventReportController.scala:55]: Is this saving event or userAnswers ? 
And do we need to say toMongo explicitly ?

Comment (cyeniceri) [inline app/controllers/EventReportController.scala:55]: This was Created before, now it's changed as OK.
Do we have a standart in HMRC ? Both are fine and understandable but I'm asking this for consistency.

File: test/controllers/EventReportControllerSpec.scala [hunk 4]
```diff
@@ -280,35 +280,95 @@ class EventReportControllerSpec extends AsyncWordSpec with Matchers with Mockito
...
+      val result = controller.getUserAnswersFromCache(fakeRequest.withHeaders(
+        newHeaders = "pstr" -> pstr, "eventType" -> eventType))
+
+      status(result) mustBe OK
+    }
+
+    "return not found exception when invalid event type" in {
+      val controller = application.injector.instanceOf[EventReportController]
+
+      when(mockEventReportService.getEventFromMongo(
```
Comment (cyeniceri) [inline test/controllers/EventReportControllerSpec.scala:302]: This mock is useless here.
++ 
Adding mockEventReportService to reset and verify that this service never called because eventType is invalid - is truer I think. 
  ```
before {
    reset(mockAuthConnector, mockJSONPayloadSchemaValidator, mockEventReportService)
    when(mockAuthConnector.authorise[Option[String]](any(), any())(any(), any())) thenReturn Future.successful(Some("Ext-137d03b9-d807-4283-a254-fb6c30aceef1"))
    when(mockJSONPayloadSchemaValidator.validateJsonPayload(any(), any())) thenReturn Right(true)
  }
```
   ```
 "return not found exception when invalid event type" in {
      val controller = application.injector.instanceOf[EventReportController]

      recoverToExceptionIf[NotFoundException] {
        controller.getUserAnswersFromCache(fakeRequest.withHeaders(
          newHeaders = "pstr" -> pstr, "eventType" -> "test"))
      } map { response =>
        verify(mockEventReportService, never).getEventFromMongo(any(), any())(any())
        response.responseCode mustBe NOT_FOUND
        response.message must include("Bad Request: eventType (test) not found")
      }
    }
```

Comment (cyeniceri) [reply test/controllers/EventReportControllerSpec.scala:302]: we should also add one more test case for valid eventType and pstr but service will return None to us.

      when(mockEventReportService.getEventFromMongo(
        ArgumentMatchers.eq(pstr),
        ArgumentMatchers.eq(Event1)
      )(any()))
        .thenReturn(Future.successful(None))

File: test/controllers/EventReportControllerSpec.scala [hunk 5]
```diff
@@ -280,35 +280,95 @@ class EventReportControllerSpec extends AsyncWordSpec with Matchers with Mockito
...
 
-      status(result) mustBe CREATED
+      status(result) mustBe OK
     }
 
     "return not found exception when invalid event type" in {
       val controller = application.injector.instanceOf[EventReportController]
 
-      when(mockEventReportService.saveEvent(
+      when(mockEventReportService.saveEventToMongo(
```
Comment (cyeniceri) [inline test/controllers/EventReportControllerSpec.scala:363]: same as above (mock is useless)

**Review by cyeniceri — CHANGES_REQUESTED** [review]

File: conf/app.routes (hunk 1 above)
Comment (grantrobson) [reply conf/app.routes]: I have renamed

File: conf/app.routes (hunk 2 above)
Comment (grantrobson) [reply conf/app.routes]: I have renamed

File: app/controllers/EventReportController.scala (hunk 3 above)
Comment (grantrobson) [reply app/controllers/EventReportController.scala]: Renamed method to make clear

Comment (grantrobson) [reply app/controllers/EventReportController.scala]: OK because multiple saves possible
